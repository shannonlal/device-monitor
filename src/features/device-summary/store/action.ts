import { Action } from 'redux';
import { IDeviceDetailsModel } from '../../../interfaces/models';

// Actions Constants
export const GET_DEVICE_DETAILS_START = 'GET_DEVICE_DETAILS_START';
export const GET_DEVICE_DETAILS_SUCCESS = 'GET_DEVICE_DETAILS_START_SUCCESS';
export const GET_DEVICE_DETAILS_ERROR = 'GET_DEVICE_DETAILS_ERROR';

// Action Types
export interface IActionGetDeviceDetailsStart extends Action {
    type: 'GET_DEVICE_DETAILS_START';
}

export interface IActionGetDeviceDetailsSuccess extends Action {
    type: 'GET_DEVICE_DETAILS_START_SUCCESS';
    deviceDetails: IDeviceDetailsModel;
}

export interface IActionGetDeviceDetailsError extends Action {
    type: 'GET_DEVICE_DETAILS_ERROR';
    errorMessage: string;
}

export type DeviceDetailsActions =
    | IActionGetDeviceDetailsStart
    | IActionGetDeviceDetailsSuccess
    | IActionGetDeviceDetailsError;
// Action Functions
export function getDeviceDetailsStart(): IActionGetDeviceDetailsStart {
    return {
        type: GET_DEVICE_DETAILS_START,
    };
}

export function getDeviceDetailsSuccess(deviceDetails: IDeviceDetailsModel): IActionGetDeviceDetailsSuccess {
    return {
        type: GET_DEVICE_DETAILS_SUCCESS,
        deviceDetails,
    };
}

export function getDeviceDetailsError(errorMessage: string): IActionGetDeviceDetailsError {
    return {
        type: GET_DEVICE_DETAILS_ERROR,
        errorMessage,
    };
}
