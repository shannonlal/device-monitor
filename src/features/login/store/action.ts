import { Action } from 'redux';
import { IAuthorization, IUser } from '../../../interfaces/models';

// Actions Constants
export const AUTHORIZE_USER_START = 'AUTHORIZE_USER_START';
export const AUTHORIZE_USER_SUCCESS = 'AUTHORIZE_USER_SUCCESS';
export const AUTHORIZE_USER_ERROR = 'AUTHORIZE_USER_ERROR';

// Action Types
export interface IActionAuthorizeUserStart extends Action {
    type: 'AUTHORIZE_USER_START';
    auth: IAuthorization;
}

export interface IActionAuthorizeUserSuccess extends Action {
    type: 'AUTHORIZE_USER_SUCCESS';
    user: IUser;
}

export interface IActionAuthorizeUserError extends Action {
    type: 'AUTHORIZE_USER_ERROR';
    errorMessage: string;
}

export type AuthorizeUserActions = IActionAuthorizeUserStart | IActionAuthorizeUserSuccess | IActionAuthorizeUserError;

// Action Functions
export function authorizeUserStart(auth: IAuthorization): IActionAuthorizeUserStart {
    return {
        type: AUTHORIZE_USER_START,
        auth,
    };
}

export function authorizeUserSuccess(user: IUser): IActionAuthorizeUserSuccess {
    return {
        type: AUTHORIZE_USER_SUCCESS,
        user,
    };
}

export function getDeviceDetailsError(errorMessage: string): IActionAuthorizeUserError {
    return {
        type: AUTHORIZE_USER_ERROR,
        errorMessage,
    };
}
