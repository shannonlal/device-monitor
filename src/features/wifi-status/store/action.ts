import { Action } from 'redux';
import { IWifiSummary, IWifiDetails } from '../../../interfaces/models';

/************** Select Wifi  *****************/
// Actions Constants
export const SELECTED_WIFI_START = 'SELECTED_WIFI_START';
export const SELECTED_WIFI_SUCCESS = 'SELECTED_WIFI_SUCCESS';
export const SELECTED_WIFI_FAIL = 'SELECTED_WIFI_FAIL';

// Action Types
export interface IActionSelectedWifiStart extends Action {
    type: 'SELECTED_WIFI_START';
    ssid: string;
}

export interface IActionSelectedWifiSuccess extends Action {
    type: 'SELECTED_WIFI_SUCCESS';
    wifiDetails: IWifiDetails;
}

export interface IActionSelectedWifiFail extends Action {
    type: 'SELECTED_WIFI_FAIL';
    errorMessage: string;
}

export type SelectWifiActions = IActionSelectedWifiStart | IActionSelectedWifiSuccess | IActionSelectedWifiFail;

// Action Functions
export function selectWifiStart(ssid: string): IActionSelectedWifiStart {
    return {
        type: SELECTED_WIFI_START,
        ssid,
    };
}

export function selectWifiSuccess(wifiDetails: IWifiDetails): IActionSelectedWifiSuccess {
    return {
        type: SELECTED_WIFI_SUCCESS,
        wifiDetails,
    };
}

export function selectWifiFail(errorMessage: string): IActionSelectedWifiFail {
    return {
        type: SELECTED_WIFI_FAIL,
        errorMessage,
    };
}

/************** Get Avaialble  Wifi  *****************/
// Actions Constants
export const GET_AVAILABILE_WIFI_START = 'GET_AVAILABILE_WIFI_START';
export const GET_AVAILABILE_WIFI_SUCCESS = 'GET_AVAILABILE_WIFI_SUCCESS';
export const GET_AVAILABILE_WIFI_FAIL = 'GET_AVAILABILE_WIFI_FAIL';

// Action Types
export interface IActionGetAvailableWifiStart extends Action {
    type: 'GET_AVAILABILE_WIFI_START';
}

export interface IActionGetAvailableWifiSuccess extends Action {
    type: 'GET_AVAILABILE_WIFI_SUCCESS';
    wifiNetworks: IWifiSummary[];
}

export interface IActionGetAvailableWifiFail extends Action {
    type: 'GET_AVAILABILE_WIFI_FAIL';
    errorMessage: string;
}

export type GetAvailableWifiActions =
    | IActionGetAvailableWifiStart
    | IActionGetAvailableWifiSuccess
    | IActionGetAvailableWifiFail;

// Action Functions
export function getAvaialbleWifiStart(): IActionGetAvailableWifiStart {
    return {
        type: GET_AVAILABILE_WIFI_START,
    };
}

export function getAvaialbleWifiSuccess(wifiNetworks: IWifiSummary[]): IActionGetAvailableWifiSuccess {
    return {
        type: GET_AVAILABILE_WIFI_SUCCESS,
        wifiNetworks,
    };
}

export function getAvaialbleWifiFail(errorMessage: string): IActionGetAvailableWifiFail {
    return {
        type: GET_AVAILABILE_WIFI_FAIL,
        errorMessage,
    };
}
