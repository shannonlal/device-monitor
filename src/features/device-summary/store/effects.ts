import { Dispatch } from 'redux';
import { getDeviceDetailsStart, getDeviceDetailsSuccess, getDeviceDetailsError } from './action';
import { IDeviceDetailsModel } from '../../../interfaces/models';
import { getDeviceBridge } from '../../../drivers/DriverFactory';

/**
 * Will get the device data
 * @return
 */
const getDeviceData = async (): Promise<IDeviceDetailsModel> => {
    const deviceDetails: IDeviceDetailsModel = await getDeviceBridge().getDeviceInfo();

    return deviceDetails;
};

/**
 *
 */
export function getDeviceDetails() {
    return async (dispatch: Dispatch) => {
        dispatch(getDeviceDetailsStart());
        try {
            const deviceDetails: IDeviceDetailsModel = await getDeviceData();
            return dispatch(getDeviceDetailsSuccess(deviceDetails));
        } catch (err) {
            return dispatch(getDeviceDetailsError(err.message));
        }
    };
}
