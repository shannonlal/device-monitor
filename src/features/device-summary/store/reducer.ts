import { IDeviceDetailsModel } from '../../../interfaces/models';
import { DeviceDetailsActions } from './action';
export interface IDeviceDetailsState {
    state: string; // 'INIT', 'LOADING' | 'LOADED' | 'ERROR',
    deviceDetails?: IDeviceDetailsModel;
    errorMessage?: string;
}

export const initialState: IDeviceDetailsState = {
    state: 'INIT',
    errorMessage: '',
};

export function deviceDetailsReducer(
    state: IDeviceDetailsState = initialState,
    action: DeviceDetailsActions,
): IDeviceDetailsState {
    if (action.type === 'GET_DEVICE_DETAILS_START') {
        return {
            ...state,
            state: 'LOADING',
        };
    }
    if (action.type === 'GET_DEVICE_DETAILS_START_SUCCESS') {
        return {
            ...state,
            state: 'LOADED',
            deviceDetails: action.deviceDetails,
        };
    }
    if (action.type === 'GET_DEVICE_DETAILS_ERROR') {
        return {
            ...state,
            state: 'ERROR',
            errorMessage: action.errorMessage,
        };
    }
    return state;
}
