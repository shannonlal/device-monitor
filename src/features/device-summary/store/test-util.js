// test-utils.js
import React, { ReactElement, ReactNode } from "react";
import { render as rtlRender, RenderOptions} from '@testing-library/react';
import { Provider } from 'react-redux';
import {
  applyMiddleware,
  combineReducers,
  compose,
  createStore,
} from 'redux';
import { 
  initialState as reducerInitialState, 
  deviceDetailsReducer 
} from './reducer';
//import reduxThunk from 'redux-thunk';
//const composeEnhancers = compose;


//const combinedReducer = combineReducers({ deviceDetails:deviceDetailsReducer  })
//const store = createStore(deviceDetailsReducer, reducerInitialState);

import {store as testStore} from '../../../store/index'
function render(
  ui,
  {
    initialState = reducerInitialState,
    testStore,
    ...renderOptions
  } = {},
) {
  function Wrapper({children}) {
    return <Provider store={testStore}>{children}</Provider>
  }
  return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
}

/*function render(
  ui,
  {
    initialState: IDeviceDetailsState = reducerInitialState,
    
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}*/

// re-export everything
export * from '@testing-library/react'

// override render method
export { render }