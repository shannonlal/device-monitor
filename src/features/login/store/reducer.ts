import { IUser, IAuthorization } from '../../../interfaces/models';
import { AuthorizeUserActions } from './action';
export interface IAuthorizationState {
    state: string; // 'INIT', 'AUTHENTICATING' | 'AUTHENTICATED' | 'ERROR',
    auth?: IAuthorization;
    user?: IUser;
    errorMessage?: string;
}

export const initialState: IAuthorizationState = {
    state: 'INIT',
    errorMessage: '',
};

export function loginReducer(
    state: IAuthorizationState = initialState,
    action: AuthorizeUserActions,
): IAuthorizationState {
    if (action.type === 'AUTHORIZE_USER_START') {
        return {
            ...state,
            state: 'AUTHENTICATING',
            auth: action.auth,
        };
    }
    if (action.type === 'AUTHORIZE_USER_SUCCESS') {
        return {
            ...state,
            state: 'AUTHENTICATED',
            user: action.user,
        };
    }
    if (action.type === 'AUTHORIZE_USER_ERROR') {
        return {
            ...state,
            state: 'ERROR',
            errorMessage: action.errorMessage,
        };
    }
    return state;
}
