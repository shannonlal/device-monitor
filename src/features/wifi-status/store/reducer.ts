import { IWifiSummary, IWifiDetails } from '../../../interfaces/models';
import { GetAvailableWifiActions } from './action';
export interface IWifiState {
    state: string; // 'INIT', 'AUTHENTICATING' | 'AUTHENTICATED' | 'ERROR',
    wifiNetworks?: IWifiSummary[];
    errorMessage?: string;
    ssid?: string;
    selectedWifi?: IWifiDetails;
}

export const initialState: IWifiState = {
    state: 'INIT',
    errorMessage: '',
    ssid: '',
};

export function wifiReducer(state: IWifiState = initialState, action: GetAvailableWifiActions): IWifiState {
    if (action.type === 'GET_AVAILABILE_WIFI_START') {
        return {
            ...state,
            state: 'INIT',
        };
    }
    if (action.type === 'GET_AVAILABILE_WIFI_SUCCESS') {
        return {
            ...state,
            state: 'RETRIEVED',
            wifiNetworks: action.wifiNetworks,
        };
    }
    if (action.type === 'GET_AVAILABILE_WIFI_FAIL') {
        return {
            ...state,
            state: 'ERROR',
            errorMessage: action.errorMessage,
        };
    }
    return state;
}
