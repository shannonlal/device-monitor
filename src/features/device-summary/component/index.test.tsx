import React from 'react';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { useDispatch, useSelector, Provider } from 'react-redux';
import DeviceSummary from './index';
import { IDeviceDetailsModel } from '../../../interfaces/models';
import { DeviceDetailsActions } from '../store/action';
import { IDeviceDetailsState } from '../store/reducer';
import { combinedReducers } from '../../../store/index';

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
}));

const useDispatchMock = useDispatch as jest.Mock;

const dispatchResultRecorder = {} as any;
const fakeDispatch = (action: DeviceDetailsActions) => {
    console.log('Calling Fake Dispatch', action.type);
    dispatchResultRecorder[action.type] = action;
};

useDispatchMock.mockImplementation(() => fakeDispatch);

const deviceDetailsModel: IDeviceDetailsModel = {
    name: 'MyPhone',
    appVersion: '1.0.1',
    appBuild: '12.1.1',
    operatingSystem: 'Android',
    osVersion: 'OS',
    platform: 'ios',
    model: 'adadaa',
    manufacturer: 'Samsung',
    uuid: '12313231213123asdsdad',
    isVirtual: true,
};

const deviceDetailsState: IDeviceDetailsState = {
    state: 'LOADED',
    deviceDetails: deviceDetailsModel,
};
const initialState = {
    deviceDetails: deviceDetailsState,
};


const useSelectorMock = useSelector as jest.Mock;
const fakeUseSelector = () => {
    console.log('Returning state object', initialState);
    return initialState;
};
useSelectorMock.mockImplementation(() => fakeUseSelector);

function renderWithProviders(ui, { reduxState } = {}) {
    const store = createStore(combinedReducers, reduxState || initialState);
    return render(<Provider store={store}>{ui}</Provider>);
}
describe('DeviceSummary', () => {
    test('rendering Device Summary with no data', async () => {
        const { asFragment } = render(<DeviceSummary />);
        expect(asFragment()).toMatchSnapshot();
    });

    /*test('rendering Device Summary with mock response', async () => {
        try {
            const deviceDetailsModel: IDeviceDetailsModel = {
                name: 'MyPhone',
                appVersion: '1.0.1',
                appBuild: '12.1.1',
                operatingSystem: 'Android',
                osVersion: 'OS',
                platform: 'ios',
                model: 'adadaa',
                manufacturer: 'Samsung',
                uuid: '12313231213123asdsdad',
                isVirtual: true,
            };

            const deviceDetailsState: IDeviceDetailsState = {
                state: 'LOADED',
                deviceDetails: deviceDetailsModel,
            };
            const initialState = {
                deviceDetails: deviceDetailsState,
            };

            const { asFragment, getByText } = renderWithProviders(<DeviceSummary />, { initialState });

            // const field = await getByText('Samsung');
            // expect(field).toBeDefined();
            expect(getByText('Samsung')).toBeInTheDocument();
            expect(asFragment()).toMatchSnapshot();
        } catch (err) {
            console.log('ee', err);
            expect(err).toBeUndefined();
        }
    });*/
});
