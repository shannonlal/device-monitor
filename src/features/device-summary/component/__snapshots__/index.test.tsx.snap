// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceSummary rendering Device Summary with mock response 1`] = `
<DocumentFragment>
  <div
    class="ion-page"
  >
    <ion-header>
      <ion-toolbar>
        <ion-title>
          Device Summary
        </ion-title>
      </ion-toolbar>
    </ion-header>
    <ion-content>
      <ion-grid>
        <ion-row>
          <ion-col>
            <ion-text />
          </ion-col>
          <ion-col>
            <ion-text />
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              appBuild
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text />
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              appVersion
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text />
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              isVirtual
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text />
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              manufacturer
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text>
              Google Inc.
            </ion-text>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              model
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text>
              Macintosh
            </ion-text>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              operatingSystem
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text>
              mac
            </ion-text>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              osVersion
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text>
              10.15.4
            </ion-text>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              platform
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text>
              web
            </ion-text>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              uuid
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text>
              775d1a10-8077-4056-91bb-ff67217af06a
            </ion-text>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              diskFree
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text>
              1110000
            </ion-text>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              memUsed
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text>
              12211111
            </ion-text>
          </ion-col>
        </ion-row>
        <ion-row>
          <ion-col>
            <ion-text>
              diskTotal
            </ion-text>
          </ion-col>
          <ion-col>
            <ion-text>
              1212121
            </ion-text>
          </ion-col>
        </ion-row>
      </ion-grid>
    </ion-content>
  </div>
</DocumentFragment>
`;

exports[`DeviceSummary rendering Device Summary with no data 1`] = `
<DocumentFragment>
  <div
    class="ion-page"
  >
    <ion-header>
      <ion-toolbar>
        <ion-title>
          Device Summary
        </ion-title>
      </ion-toolbar>
    </ion-header>
    <ion-content>
      <ion-grid>
        <ion-row>
          <ion-col>
            <ion-text />
          </ion-col>
          <ion-col>
            <ion-text />
          </ion-col>
        </ion-row>
      </ion-grid>
    </ion-content>
  </div>
</DocumentFragment>
`;
